zhaquirks/tuya/ts0601_temperaturehumidity.py

```diff
@@ -0,0 +1,140 @@
+# Unknown message (b'19830100a40102000400000118') on cluster 61184: unknown endpoint or cluster id: 'No cluster ID 0xef00 on (a4:c1:38:d0:18:8b:64:aa, 1)'
+#                                          28.0 degrees
+# Unknown message (b'19840100a5020200040000022c') on cluster 61184: unknown endpoint or cluster id: 'No cluster ID 0xef00 on (a4:c1:38:d0:18:8b:64:aa, 1)'
+#                                          55.6% humid
+# Unknown message (b'19850100a60402000400000064') on cluster 61184: unknown endpoint or cluster id: 'No cluster ID 0xef00 on (a4:c1:38:d0:18:8b:64:aa, 1)'
+#                                          100% battery
+
+# Constants that map to the string values in the previously `Unknown message` lines.
+TUYA_TEMPERATURE_ATTR = 0x0201  # [0, 0, 0, 237] temperature in decidegree
+TUYA_HUMIDITY_ATTR = 0x0202  # [0, 0, 2, 64] humidity
+TUYA_BATTERY_ATTR = 0x0204  # [0, 0, 0, 100] battery
+
+
+class TuyaTemperatureMeasurement(TemperatureMeasurement, TuyaLocalCluster):
+    """Tuya local TemperatureMeasurement cluster."""
+
+
+class TuyaRelativeHumidity(RelativeHumidity, TuyaLocalCluster):
+    """Tuya local RelativeHumidity cluster."""
+
+
+class TemperatureHumidityManufCluster(TuyaNewManufCluster):
+    """Tuya Manufacturer Cluster with Temperature and Humidity data points."""
+
+    attributes = TuyaManufClusterAttributes.attributes.copy()
+    attributes.update(
+        {
+            TUYA_TEMPERATURE_ATTR: ("temperature", t.uint32_t, True),
+            TUYA_HUMIDITY_ATTR: ("humidity", t.uint32_t, True),
+            TUYA_BATTERY_ATTR: ("battery", t.uint8_t, True),
+        }
+    )
+
+    dp_to_attribute: Dict[int, DPToAttributeMapping] = {
+        1: DPToAttributeMapping(
+            TuyaTemperatureMeasurement.ep_attribute,
+            "measured_value",
+            dp_type=TuyaDPType.VALUE,
+            converter=lambda x: x * 10,  # decidegree to centidegree
+        ),
+        2: DPToAttributeMapping(
+            TuyaRelativeHumidity.ep_attribute,
+            "measured_value",
+            dp_type=TuyaDPType.VALUE,
+            converter=lambda x: x * 10,  # decipercent to centipercent
+        ),
+        4: DPToAttributeMapping(
+            TuyaPowerConfigurationCluster2AAA.ep_attribute,
+            "battery_percentage_remaining",
+            dp_type=TuyaDPType.VALUE,
+            converter=lambda x: x * 2,  # double reported percentage
+        ),
+    }
+
+    data_point_handlers = {
+        1: "_dp_2_attr_update",
+        2: "_dp_2_attr_update",
+        4: "_dp_2_attr_update",
+    }
+
+
+class TuyaTempHumiditySensor(CustomDevice):
+    """Custom device representing tuya temp and humidity sensor with e-ink screen."""
+
+    def __init__(self, *args, **kwargs):
+        """Init device."""
+
+        self.temperature_bus = Bus()
+        self.humidity_bus = Bus()
+        self.battery_bus = Bus()
+
+        super().__init__(*args, **kwargs)
+
```

Comment:
No need for attributes Bus since we are not triggering events.

Reply:
Removed.